### PR TITLE
try to get clang to spit out better error messages without breaking the world

### PR DESCRIPTION
**Pull request by ericniebler**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -84,7 +84,12 @@
         "__split_buffer": "cpp",
         "filesystem": "cpp",
         "queue": "cpp",
-        "stack": "cpp"
+        "stack": "cpp",
+        "__errc": "cpp",
+        "__hash_table": "cpp",
+        "__string": "cpp",
+        "algorithm": "cpp",
+        "codecvt": "cpp"
```
Comment (CaseyCarter) [inline .vscode/settings.json]: You really should put a `"**/include/c++/**": "cpp",` line in here so VSCode stops spamming every libc++ header here.

Comment (CaseyCarter) [reply .vscode/settings.json]: Also, ewww `<codecvt>`.

**Comment by CaseyCarter** [comment]
This passes CI after merging my change from master, but it's still broken in non-`/std:c++latest` mode (Appveyor only covers `/std:c++latest` with cl). I suggest that you merge anyway.

**Comment by ericniebler** [comment]
Interestingly, something about this PR improved compile times according to Travis. 🤷‍♂

**Pull request merged**